Add System*Native libraries from .NET Core 3.1 release, to Platform Manifest

NikolaMilosavljevic opened the pull request:
Platform manifest needs to include all System*Native items from 3.1 release. We need to add them explicitly, to work around the issue introduced with a fix for https://github.com/dotnet/runtime/issues/33118 - Android libraries need to conform to a specific naming convention which necessitated renames of these libraries.

List of libraries being added:
```
System.IO.Compression.Native.a
System.IO.Compression.Native.so
System.Native.a
System.Native.so
System.Net.Http.Native.a
System.Net.Http.Native.so
System.Net.Security.Native.a
System.Net.Security.Native.so
System.Security.Cryptography.Native.OpenSsl.a
System.Security.Cryptography.Native.OpenSsl.so
System.Globalization.Native.so
System.IO.Compression.Native.dylib
System.Native.dylib
System.Net.Http.Native.dylib
System.Net.Security.Native.dylib
System.Security.Cryptography.Native.Apple.a
System.Security.Cryptography.Native.Apple.dylib
System.Security.Cryptography.Native.OpenSsl.dylib
System.Globalization.Native.dylib
```

Fixes the issue: https://github.com/dotnet/runtime/issues/33450

NikolaMilosavljevic committed:
Add System*Native libraries from .NET Core 3.1 release, to Platform Manifest

NikolaMilosavljevic committed:
Add System*Native libraries from .NET Core 3.1 release, to Platform Manifest

ericstj commented on src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj:
```diff
@@ -70,6 +70,42 @@
+      Android libraries need to conform to a specific naming convention which necessitated renames of these libraries.
+      For more details of the issue and fix see https://github.com/dotnet/runtime/issues/33450 -->
+
+  <Target Name="AddSharedFrameworkNativeFilesWithoutLibPrefixToPlatformManifest"
```
Why are you using a target for this?  Can't this all be declared statically?

Maybe something like this:
```
<ItemGroup>
  <_pastShimFilesNames Include="System.Native;System.Net.Http.Native;System.Security.Cryptography.Native.OpenSsl;System.Gobalization.Native" />
  <_pastShimFiles Include="@(_pastShimFilesNames->'%(Identity).so')" />
  <_pastShimFilesNames Include="System.Security.Cryptography.Native.Apple" />
  <_pastShimFiles Include="@(_pastShimFilesNames->'%(Identity).a')" />
  <_pastShimFiles Include="@(_pastShimFilesNames->'%(Identity).dylib')" />
  <SharedFrameworkRuntimeFiles Include="@(_pastShimFiles)" TargetPath="runtimes/" />
</ItemGroup>
```

NikolaMilosavljevic replied:
A target would ensure these entries were added to `SharedFrameworkRuntimeFiles` before it's consumed in `GenerateFileVersionProps`

dagood replied:
The project evaluation phase completes before any task is run. (I can't find a page that specifically mentions this about evaluation, but https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2019#create-items-during-execution comes very close.) I agree putting them down as static items is better if it works, here.

(Generating the names based on patterns is also nice to see the intent rather than a simple list.)

NikolaMilosavljevic replied:
Cool, thanks! I will update the PR to incorporate this suggested change.

monojenkins committed:
[jit] Perform inlining earlier during call processing to avoid loading vtable/rgctx args which are not needed. (#33951)

Co-authored-by: vargaz <vargaz@users.noreply.github.com>

dagood commented on src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj:
```diff
@@ -70,6 +70,42 @@
+      to work around the issue introduced with a fix for https://github.com/dotnet/runtime/issues/33118,
+      Android libraries need to conform to a specific naming convention which necessitated renames of these libraries.
+      For more details of the issue and fix see https://github.com/dotnet/runtime/issues/33450 -->
+
```
(Nit) Remove empty line to make doc comment adjacent?

My style personally is this, which incorporates a readability gap:

```xml
  <!--
    Platform manifest needs to include all System*Native items from 3.1 release. This target adds
    them explicitly, to work around the issue introduced with a fix for
    https://github.com/dotnet/runtime/issues/33118, Android libraries need to conform to a specific
    naming convention which necessitated renames of these libraries. For more details of the issue
    and fix see https://github.com/dotnet/runtime/issues/33450
  -->
  <ItemGroup>
```

dagood commented on src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj:
```diff
@@ -70,6 +70,42 @@
 
+  <!-- Platform manifest needs to include all System*Native items from 3.1 release. This target adds them explicitly,
+      to work around the issue introduced with a fix for https://github.com/dotnet/runtime/issues/33118,
+      Android libraries need to conform to a specific naming convention which necessitated renames of these libraries.
```
(Nit) Extra clarity

```suggestion
  <!-- Platform manifest needs to include all System*Native items from 3.1 release. This target adds them explicitly,
      to fill the gap that appeared when the
      names of these libraries were changed in 5.0 to conform to a specific Android naming convention. The library name change was a fix for https://github.com/dotnet/runtime/issues/33118.
```

(Suggestion feature doesn't handle newlines nicely, so couldn't format it.)

NikolaMilosavljevic committed:
Update src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj

Co-Authored-By: Davis Goodin <dagood@users.noreply.github.com>

0xfk0 committed:
Fixed build with GDBJIT enabled and PREJIT disabled (#33958)

This commit fixes build for case, when FEATURE_GDBJIT
is enabled, but FEATURE_PREJIT is disabled.

This solves issue #33956.

jkotas committed:
More efficient interop for BITMAPINFOHEADER (#33967)

lambdageek committed:
[mono] Add headers for unstable APIs (#33869)

* [mono] Add headers for unstable APIs

We have functions that are de-facto part of the Mono embedding API that are
used by Xamarin.iOS or Xamarin.Android that are not in a state where we want to
commit to supporting them forever in their current form.

Nonetheless, we would like to make sure that the symbols for these functions
are visible even if we otherwise compile the runtime with
`-fvisisibility=hidden`, and we would like to discourage declaring the
functions in the embedder's own headers.

The functions that go into the
`mono/{utils,metadata,jit}/mono-private-unstable.h` headers will all be marked
with `MONO_API MONO_RT_EXTERNAL_ONLY` but we will not guarantee that the
functions will be there from one release to the next or that they will not
change their signatures or their behaviors.

* [msvc] Add the three mono-private-unstable.h files

NikolaMilosavljevic committed:
Removed targets, renamed item-list and sorted the list.

NikolaMilosavljevic committed:
Merge branch 'sys.native' of https://github.com/NikolaMilosavljevic/runtime into sys.native

NikolaMilosavljevic commented:
Abandoning due to issue merge in my latest commit - will send a new one.
